Firestore/core/src/firebase/firestore/util/status.h

```diff
@@ -100,6 +100,7 @@ class ABSL_MUST_USE_RESULT Status {
   /// \brief Return a string representation of this status suitable for
   /// printing. Returns the string `"OK"` for success.
   std::string ToString() const;
+  friend std::ostream& operator<<(std::ostream& out, const Status& status);
```

Comment:
The definition doesn't actually use any private fields. I think you can just drop this friend declaration.

Reply:
My goal is to declare these near the `ToString()` to make it obvious which classes support this usage and which don't yet. I want to make this boilerplate something you can mostly copypasta into the next class.

In addition to giving access to internals `friend` declares a free function but inside the body of the class. There's no equivalent way to declare this operator without moving it outside the class, which is annoying.

See: https://stackoverflow.com/a/2315264/6253640